neo4j/src/main/java/org/hibernate/ogm/datastore/neo4j/transaction/impl/Neo4jTransactionCoordinatorBuilder.java

```diff
@@ -30,11 +31,14 @@ public Neo4jTransactionCoordinatorBuilder(TransactionCoordinatorBuilder delegate
 
 	@Override
 	public TransactionCoordinator buildTransactionCoordinator(TransactionCoordinatorOwner owner, TransactionCoordinatorOptions options) {
-		TransactionCoordinator coordinator = delegate.buildTransactionCoordinator( owner, options );
 		if ( delegate.isJta() ) {
+			TransactionCoordinator coordinator = delegate.buildTransactionCoordinator( owner, options );
 			return new Neo4jJtaTransactionCoordinator( coordinator, datastoreProvider );
 		}
 		else {
+			// We use a no-op owner to make sure that flushes and events are triggered as expected.
+			NoopJdbcResourceTransactionCoordinatorOwner noopOwner = new NoopJdbcResourceTransactionCoordinatorOwner( owner );
```

Comment:
@gunnarmorling I force-pushed an alternative solution for this. I'm not using anymore the No-op impl anymore and I've created an implementation for Neo4j.

Does that make more sense, now?